src/it/java/org/jboss/tools/intellij/openshift/test/ui/AbstractBaseTest.java

```diff
@@ -32,20 +34,26 @@
 public abstract class AbstractBaseTest {
 
     protected static RemoteRobot robot;
+    private static boolean hasConnectedToTestIDE = false;
+    public static final String DEFAULT_CLUSTER_URL = "no (current) context/cluster set";
 
     @BeforeAll
     public static void connect() {
-        robot = IdeaRunner.getInstance().getRemoteRobot();
-        ProjectUtility.createEmptyProject(robot, "test-project");
+        if (!hasConnectedToTestIDE) {
+            robot = IdeaRunner.getInstance().getRemoteRobot();
+            hasConnectedToTestIDE = true;
```

Comment:
What is the reason for setting hasConnectedToTestIDE to false and checking its value with condition? The condition always evaluate as True, so it seems a little bit unnecessary.

Reply:
@lgrossma I intend the individual Public or CLuster tests to inherit from these abstract classes:

```
AbstractBaseTest
├── AbstractPublicTest
│   ├── TestClass1 (Public Tests)
│   ├── TestClass2 (Public Tests)
│   └── ...
└── AbstractClusterTest
    ├── TestClass1 (Cluster Tests)
    ├── TestClass2 (Cluster Tests)
    └── ...

```
With this setup we only need the **first Test** to get RemoteRobot instance and create a project in intellij. From that point we can test the extension without every individual test to close and create project and the remote robot instance is common across all of the tests. 

So, to answer the question. The **_hasConnectedToTestIDE_** flag ensures that the test-project is created only once no matter which test is ran first. But WaitUntilBigTasksFinish runs before every test, so any code needed to run before each test would fall here.

I am not sure about the name of the flag or the connect method. Maybe refactor to setup before tests or something along those lines? (it was already there) Thoughts or suggestions?